clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+The app is running in the background after it is activated by MVA, so it cannot display UI. For an audio-only app interaction, your application can simply stay in the background, play audio output, and listen to audio input. To display UI, call ConversationalAgentSession.RequestForegroundActivationAsync. In the sample app, this is called in [App.xaml.cs](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/App.xaml.cs)'s OnSignalConfirmed which is triggered by the SignalConfirmed event in DialogManager.
+
+### Summary of activation in the UWP Voice Assistant Sample
+When the app is started, the sample app 
+- verifies that an audio input device is available and accessible
+- registers itself with the Background Service
+- registers its keyword
+- verifies the voice activation is enabled
+
+When the sample app receives an activation from MVA, the sample app
+- retrieves an audio buffer that contains the "activation audio", including audio since 3 seconds before the 1st stage keyword detection
+- provides the audio buffer as an input to the Speech SDK for 2nd stage keyword verification (and starting a timer in case of keyword rejection)
+- reacts to keyword verification succeeding or failing by surfacing corresponding events.
+
+Code path in the sample app:
+- On first run MainPage.xaml.cs completes keyword registration, Background Service registration, mic verification, and voice activation setting verification.
+- On 1st stage detection, DialogManager.HandleSignalDetected is called, either through App.xaml.cs's OnBackgroundActivated or the ConversationalAgentSession.SignalDetected event handler in AgentSessionManager
+- HandleSignalDetected initializes the AgentAudioInputProvider with audio from the ConversationalAgentSession, creates a DirectLineSpeechDialogBackend instance, and passes the AgentAudioInputProvider through SetAudioSource
+- DialogManager listens for speech recognition events from the DirectLineSpeechDialogBackend or a timeout event from SignalDetectionHelper and surfaces the events to the UI
+- On keyword confirmation, the app uses the ConversationalAgentSession to request foreground activation and display UI
+
+## Bot interaction
+The IDialogBackend interface was designed to allow the use of any dialog service with the other components in the UWP Voice Assistant sample. It contains events for keyword signals, speech recognition output, and generic output in the form of the DialogResponse class. 
+
+The sample app's implementation of this interface, DirectLineSpeechDialogBackend, uses the Speech SDK and Direct Line Speech which provide an easy way to build a dialog service and a straightforward library to interface with it.
+
+The following is a walkthrough of the code path to process an utterance from the user while the app is listening followed by a text and audio response from the bot, altogether referred to as a "turn".
```

Comment:
Not for this PR: we likely want to split this part (and maybe some of the others) out into more bite-sized .md files. This is an awesome collection of topics that's worthy of some extra navigation niceties. Can and should come later, though.